### PR TITLE
Add golangci-lint for more linting goodness

### PR DESCRIPTION
**Pull request by ddworken**
Per @joshblum's suggestion, adds golangci-lint. This adds it in both the pre-commit hooks and on circle.

### CODE REVIEW

File: src/keybaseca/sshutils/sshutils.go [hunk 1]
```diff
@@ -92,6 +92,9 @@ func ProcessSignatureRequest(conf config.Config, sr shared.SignatureRequest) (re
 	log.Log(conf, fmt.Sprintf("Processing SignatureRequest from user=%s on device='%s' keyID:%s, principals:%s, expiration:%s, pubkey:%s",
 		sr.Username, sr.DeviceName, keyID, principals, conf.GetKeyExpiration(), sr.SSHPublicKey))
 	signature, err := SignKey(conf.GetCAKeyLocation(), keyID, principals, conf.GetKeyExpiration(), sr.SSHPublicKey)
+	if err != nil {
```
Comment (ddworken) [inline src/keybaseca/sshutils/sshutils.go]: This was actually a really good catch from it :)

File: .circleci/config.yml [hunk 2]
```diff
@@ -24,6 +24,8 @@ jobs:
       - run: go get golang.org/x/tools/cmd/goimports
       - run: "! goimports ./... 2>&1 | read"
       - run: "! go mod tidy ./... 2>&1 | read"
+      - run: go get -u github.com/golangci/golangci-lint/cmd/golangci-lint
```
Comment (joshblum) [inline .circleci/config.yml]: can you pin to a version? client is on `1.16.0` https://github.com/keybase/client/blob/master/Jenkinsfile#L463, https://github.com/golangci/golangci-lint#ci-installation

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+linters:
```
Comment (joshblum) [inline .golangci.yml]: can you mirror the client settings: https://github.com/keybase/client/blob/master/go/.golangci.yml#L5-L15

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -13,3 +13,7 @@ repos:
     rev: f16f18866390cba642e8af6073ed59504aeada5c
     hooks:
     -   id: go-mod-tidy
+-   repo: https://github.com/golangci/golangci-lint
+    rev: d2b1eea2c6171a1a1141a448a745335ce2e928a1
```
Comment (joshblum) [inline .pre-commit-config.yaml]: can you change to rev for `1.16.0`

File: src/cmd/kssh/kssh.go [hunk 5]
```diff
@@ -235,13 +235,14 @@ func getConfig(botname string) (kssh.ConfigFile, error) {
...
 	configs, botnames, err := kssh.LoadConfigs()
 	if err != nil {
-		return empty, fmt.Errorf("Failed to load config file(s): %v\n", err)
+		return empty, fmt.Errorf("Failed to load config file(s): %v", err)
 	}
-	if len(configs) == 0 {
+	switch len(configs) {
+	case 0:
 		return empty, fmt.Errorf("Did not find any config files in KBFS (is `keybaseca service` running?)")
-	} else if len(configs) == 1 {
```
Comment (joshblum) [inline src/cmd/kssh/kssh.go]: up to you, but you can revert this if you take the client config settings. this rule was voted out

File: .pre-commit-config.yaml (hunk 4 above)
Comment (ddworken) [reply .pre-commit-config.yaml]: Done by just adding it from the keybase repo.

File: .circleci/config.yml (hunk 2 above)
Comment (ddworken) [reply .circleci/config.yml]: Done

File: .golangci.yml (hunk 3 above)
Comment (ddworken) [reply .golangci.yml]: Mirrored except stricter. There are some additional linters that I have enabled that seem useful and are more practical to have enabled given that this is a smaller project. Do you object to any of the additional ones I added?

File: src/cmd/kssh/kssh.go (hunk 5 above)
Comment (ddworken) [reply src/cmd/kssh/kssh.go]: /shrug I like that change so I'll leave it

File: .golangci.yml (hunk 3 above)
Comment (joshblum) [reply .golangci.yml]: sounds great!

**Pull request merged**